src/lib/components/Text.tsx

```diff
@@ -30,11 +32,20 @@ const Text: React.FunctionComponent<ITextProps & IConsumerInjectedProps> = ({
     },
   })
 
-  return (
+  const textElement = () => (
     <OText style={appliedStyle.style} {...rest}>
       {getTextChildren(rest, children as string)}
     </OText>
   )
+
+  // Go through all directives before render
+  const renderWithDirectives = () => {
```

Comment:
I'd say you should have this logic inside the util folder, preferably in `src/lib/utils/directives.tsx`. This will make it much cleaner and less redundant. This file would look something like this:

```
const element = () => (
    <OText style={appliedStyle.style} {...rest}>
      {getTextChildren(rest, children as string)}
    </OText>
  )

return renderWithDirectives(element)
```

If you decide to do this, please don't forget to do the same inside the view component as well.

Reply:
Good idea 👍 I'll change it